src/internal/Thresholding.cpp

```diff
@@ -0,0 +1,23 @@
+#include <internal/Thresholding.h>
+
+Thresholding::Thresholding (double amplitude,double digital_zero,double digital_one):
+  m_digital_zero(digital_zero)
+  , m_digital_one(digital_one)
+{
+}
+
+
+void Thresholding::setThreshold (double threshold){
+  m_threshold = threshold;
+}
+
+double Thresholding::getThreshold (void){
+  return m_threshold;
+}
+
+double Thresholding::getThresholdSignal (double signal){
+  if (signal > m_threshold)
```

Comment:
This code is good. But sometimes you can reduce this using a ternary operator. See below:

```c++
return (signal > m_threshold) ? m_digital_one : m_digital_zero;
```
If you don't want to use ternary operator, you can remove the `else`:

```c++
if (signal > m_threshold)
  return m_digital_one;

return m_digital_zero;
```

Reply:
agree

Reply:
I like the ternary, there is some in other files.